modules/tinymce/src/themes/silver/test/ts/browser/view/ViewButtonsContextTest.ts

```diff
@@ -0,0 +1,214 @@
+              }
+            ]
+          },
+          {
+            type: 'group',
+            buttons: [
+              { type: 'button', text: 'Cancel', onAction: store.adder('Cancel') },
+              { type: 'button', text: 'Save code', buttonType: 'primary', onAction: store.adder('Save code') }
+            ]
+          },
+        ],
+        onShow: (api: any) => {
+          api.getContainer().innerHTML = '<button>myview1</button>';
+          api.getContainer().querySelector('button')?.focus();
+        },
+        onHide: Fun.noop
+      });
+
+      editor.ui.registry.addContextToolbar('test-context', {
+        predicate: (node) => node.nodeName.toLowerCase() === 'img',
+        items: 'bold'
+      });
+    }
+  }, []);
+
+  const toggleView = (name: string) => {
+    const editor = hook.editor();
+    editor.execCommand('ToggleView', false, name);
+  };
+
+  const clickViewButton = (editor: Editor, tooltip: string) => TinyUiActions.clickOnUi(editor, `.tox-view button[aria-label='${tooltip}']`);
+
+  const getSvg = (editor: Editor, name: string) => UiFinder.findIn<HTMLElement>(TinyDom.container(editor), `.tox-view button[aria-label='${name}'] svg`).getOrDie().dom.innerHTML;
+
+  const getButtonByTitle = (title: string) => {
+    const editor = hook.editor();
+    return UiFinder.findIn<HTMLElement>(TinyDom.container(editor), `.tox-view button[aria-label='${title}']`).getOrDie();
+  };
+
+  it('TINY-11211: View toggle button context should reflect button state', () => {
+    const editor = hook.editor();
+
+    toggleView('myview1');
+    editor.mode.set('testmode');
+
+    assertButtonNativelyEnabled('button-with-toggle');
+    const initialButtonWithToggleButtonSvg = getSvg(editor, 'button-with-toggle');
+    clickViewButton(editor, 'button-with-toggle');
+    assert.notEqual(getSvg(editor, 'button-with-toggle'), initialButtonWithToggleButtonSvg, 'after the first toggle icon should change');
+    clickViewButton(editor, 'button-with-toggle');
+    assert.equal(getSvg(editor, 'button-with-toggle'), initialButtonWithToggleButtonSvg, 'after the second toggle icon should return to the old value');
+    store.sAssertEq('Clicking on button should should be permitted', [ 'button-with-toggle' ]);
+
+    assertButtonNativelyEnabled('button-without-toggle');
+    const initialbuttonWithoutToggleButtonSvg = getSvg(editor, 'button-without-toggle');
+    clickViewButton(editor, 'button-without-toggle');
+    assert.equal(getSvg(editor, 'button-without-toggle'), initialbuttonWithoutToggleButtonSvg, 'click should not toggle icon');
+    clickViewButton(editor, 'button-without-toggle');
+    assert.equal(getSvg(editor, 'button-without-toggle'), initialbuttonWithoutToggleButtonSvg, 'click should not toggle icon');
+    store.sAssertEq('Clicking on button should should be permitted', [ 'button-with-toggle', 'button-without-toggle' ]);
+  });
+
+  it('TINY-9616: if is active is true the button should have ViewButtonClasses.Ticked', async () => {
```

Comment:
I think update these to the new ticket ID for this test since these are all new tests that check the context even though they are duplicated. This test in particular doesn't seem to do anything with modes should we remove it.

Reply:
Ah 😞 fixing them, forgot to double check before pushing them.